src/unix/lwt_unix_unix.h

```diff
@@ -3008,6 +3009,41 @@ CAMLprim value lwt_unix_bind_job(value fd, value address)
     return lwt_unix_alloc_job(&job->job);
 }
 
+/* getcwd */
+
+/*
+In the OCaml sources, getcwd is set as either
+- 4096 (in asmrun/spacetime.c, byterun/sys.c)
+- PATH_MAX, MAXPATHLEN, or 512 (in otherlibs/unix/getcwd.c)
+*/
+
+struct job_getcwd {
+    struct lwt_unix_job job;
+    char buf[4096];
+    char* result;
+    int error_code;
+};
+
+static void worker_getcwd(struct job_getcwd *job)
+{
+    job->result = getcwd(job->buf, sizeof(job->buf));
+    job->error_code = errno;
+}
+
+static value result_getcwd(struct job_getcwd *job)
+{
+    LWT_UNIX_CHECK_JOB(job, job->result == NULL, "getcwd");
+    value result = caml_copy_string(job->result);
+    lwt_unix_free_job(&job->job);
+    return result;
+}
+
+CAMLprim value lwt_unix_getcwd_job()
```

Comment:
This function should take an argument `value unit` (see elsewhere in the file). Technically, it's not necessary, since this is C and we don't try to use the argument, but OCaml is passing that argument to this function when it calls it (and its value is zero, though it may be encoded as 1, I forgot how `()` is actually encoded).